src/contour/TerminalSession.cpp

```diff
@@ -833,7 +833,8 @@ void TerminalSession::sendKeyEvent(Key key, Modifiers modifiers, KeyboardEventTy
         if (auto const* actions =
                 config::apply(_config.inputMappings.value().keyMappings, key, modifiers, matchModeFlags()))
         {
-            executeAllActions(*actions);
+            if (eventType != KeyboardEventType::Repeat)
```

Comment:
instead of adding the `if` here, change the already existing `if`-condition in line 831 to `== KeyboardEventType::Press`.

Reply:
Updated

Reply:
Now it will make a bell sound while holding key, but i suppose this is ok

Reply:
> Now it will make a bell sound while holding key, but i suppose this is ok

I think I am confused. :thinking:

Reply:
Previously, during `Repeat` event type, input was ignored and nothing was happening, after suggested changes we ignore this branch for `Repeat` events and send them further into the terminal, and since vt backend does not know what to do with them it notifies user that something is happening via bell sound

Reply:
you're right. Repeat should only be sent if a Press was sent out first. So I think the solution is a little bit more complex:

If an input binding matches, it'll be considered "handled". But only Press is going to be sent.
Anything else is forwarded as usual.

But maybe we could leave the logic as-is and only check for Press event (vs Repeat) inside NewTab action. Because most actions you probably want to be spammable. (I at least checked with konsole TE, they only guard against spamming for spawning a new tab)

Reply:
I update it and now we trigger action only on press events, also i think we can keep all actions triggered only on press events